packages/api/src/task/task.ts

```diff
@@ -84,21 +88,65 @@ export default class TaskScheduler {
             status: "ready",
           });
 
-          await db.task.update(task.id, {
-            status: {
-              phase: "completed",
-              updatedAt: Date.now(),
-            },
-            output: event.output,
-          });
           return true;
         }
       }
-      console.log(`task type unknown: ${event.task.type}`);
+
+      if (allowedTasks.includes(event.task.type)) {
+        await db.task.update(task.id, {
+          status: {
+            phase: "completed",
+            updatedAt: Date.now(),
+          },
+          output: event.output,
+        });
+        return true;
+      } else {
+        console.log(`task type unknown: ${event.task.type}`);
+      }
     } else {
       console.log(`task event process error: task ${event.task.id} not found`);
       return true;
     }
     return false;
   }
+
+  async scheduleTask(
+    asset: Asset,
+    type: "import" | "export" | "transcode",
+    params: object
+  ) {
+    let newTask: WithID<Task> = {
+      id: uuid(),
+      name: `asset-upload-${asset.name}-${asset.createdAt}`,
+      createdAt: Date.now(),
+      type: type,
+      parentAssetId: asset.id,
+      userId: asset.userId,
+      params: {},
+      status: {
+        phase: "pending",
+        updatedAt: Date.now(),
+      },
+    };
+
+    newTask.params[type] = params;
+
+    let task = await db.task.create(newTask);
+    await this.queue.publish("task", `task.trigger.${task.type}.${task.id}`, {
+      type: "task_trigger",
+      id: uuid(),
+      timestamp: Date.now(),
+      task: {
+        id: task.id,
+        type: task.type,
+        snapshot: task,
+      },
+    });
+    await db.task.update(task.id, {
+      status: { phase: "waiting", updatedAt: Date.now() },
+    });
```

Comment:
Oh you do already have that here! Please remove the repeated update on every place you call `scheduleTask`.